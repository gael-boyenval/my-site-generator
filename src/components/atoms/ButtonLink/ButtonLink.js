import React from 'react';
import PropTypes from 'prop-types';
import { TinyHeading } from '../Typography';
import Icon from '../Icon';
import { colors, unit } from '../../../styles/tokens';
import { rem } from '../../../styles/utils';
import { Margin } from '../../layout/Spaces';

const Button = TinyHeading.withComponent('a').extend`
  color: ${colors.electric};
  line-height: ${rem(unit)};
  display: inline-flex;
  margin: 0;
`;

const ButtonLink = ({ children, ...rest }) => (
  <Button {...rest}>
    <Icon name="arrowRight" size={rem(unit)} inlineBlock color={colors.electric} />
    <Margin left="s" inline>
      {children}
    </Margin>
  </Button>
);

ButtonLink.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ButtonLink;
