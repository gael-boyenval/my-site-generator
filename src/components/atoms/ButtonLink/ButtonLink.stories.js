/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import ButtonLink from './ButtonLink';

storiesOf('01 - Atoms', module).add('ButtonLink', withInfo()(() => <ButtonLink>hello</ButtonLink>));
