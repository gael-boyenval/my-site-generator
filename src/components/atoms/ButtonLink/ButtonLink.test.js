/* eslint-disable no-undef */
import React from 'react';

import ButtonLink from './ButtonLink';

describe('Component ButtonLink', () => {
  it('should be render without crash', () => {
    const component = shallow(<ButtonLink>hello</ButtonLink>);

    expect(toJson(component)).toMatchSnapshot();
  });
});
