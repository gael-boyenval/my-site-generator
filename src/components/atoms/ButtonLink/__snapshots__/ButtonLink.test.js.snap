// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ButtonLink should be render without crash 1`] = `
<styled.a>
  <Icon
    color="#62E0BB"
    fill="none"
    inlineBlock={true}
    name="arrowRight"
    size="1.2rem"
    strokeWidth="1px"
  />
  <styled.div
    all={false}
    bottom={false}
    horizontal={false}
    inline={true}
    left="s"
    right={false}
    top={false}
    vertical={false}
  >
    hello
  </styled.div>
</styled.a>
`;
